### PR TITLE
Various examples updates.

### PR DESCRIPTION
**Pull request by anntzer**
- don't import out of pyplot or numpy, import as plt/np.
- move some examples to use subplots() instead of add_subplot.
- set random seed for some examples.
- fix some docstrings.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/event_handling/zoom_window.py [hunk 1]
```diff
@@ -6,27 +6,27 @@
...
+the x,y coordinates of the clicked point.
 
-Note the diameter of the circles in the scatter are defined in
-points**2, so their size is independent of the zoom
+Note the diameter of the circles in the scatter are defined in points**2, so
+their size is independent of the zoom.
 """
-from matplotlib.pyplot import figure, show
+
+from matplotlib import pyplot as plt
```
Comment (timhoffm) [inline examples/event_handling/zoom_window.py]: `import matplotlib.pyplot as plt` (also in various other locations, please search/replace)

File: examples/units/evans_test.py [hunk 2]
```diff
@@ -75,24 +75,18 @@ def default_units(x, axis):
...
-ax.plot(x, y, 'o', xunits=2.0)
-for label in ax.get_xticklabels():
-    label.set_rotation(30)
-    label.set_ha('right')
-ax.set_title("xunits = 2.0")
 
+# plot specifying units
+ax2.plot(x, y, 'o', xunits=2.0)
+ax2.set_title("xunits = 2.0")
+plt.setp(ax2.get_xticklabels(), rotation=30, ha='right')
```
Comment (timhoffm) [inline examples/units/evans_test.py]: Broadcasting explicitly to a list of tick labels feels a bit clumsy. Is there maybe a method to set this for all tick labels, e.g. `tick_params` or similar?

File: examples/units/evans_test.py [hunk 3]
```diff
@@ -75,24 +75,18 @@ def default_units(x, axis):
...
 # plot without specifying units; will use the None branch for axisinfo
-ax = fig.add_subplot(1, 2, 1)
-ax.plot(x, y)  # uses default units
-ax.set_title('default units')
-for label in ax.get_xticklabels():
-    label.set_rotation(30)
-    label.set_ha('right')
+ax1.plot(x, y)  # uses default units
+ax1.set_title('default units')
+plt.setp(ax1.get_xticklabels(), rotation=30, ha='right')
```
Comment (timhoffm) [inline examples/units/evans_test.py]: see above

File: examples/user_interfaces/embedding_in_gtk_sgskip.py [hunk 4]
```diff
@@ -25,8 +25,8 @@
 
 f = Figure(figsize=(5, 4), dpi=100)
 a = f.add_subplot(111)
-t = arange(0.0, 3.0, 0.01)
-s = sin(2*pi*t)
+t = np.arange(0.0, 3.0, 0.01)
+s = np.sin(2*np.pi*t)
```
Comment (timhoffm) [inline examples/user_interfaces/embedding_in_gtk_sgskip.py]: Do we have a convention, if this should include spaces around the operators? PEP8 would want them. Throughout the examples both variants are used.

File: examples/axes_grid1/demo_colorbar_with_axes_divider.py [hunk 5]
```diff
@@ -4,27 +4,23 @@
...
+
 import matplotlib.pyplot as plt
 from mpl_toolkits.axes_grid1.axes_divider import make_axes_locatable
-
 from mpl_toolkits.axes_grid1.colorbar import colorbar
-# from matplotlib.pyplot import colorbar
 
-fig = plt.figure(1, figsize=(6, 3))
+fig, (ax1, ax2) = plt.subplots(1, 2)
 fig.subplots_adjust(wspace=0.5)
```
[the remark this answers is not in the document]
Comment (anntzer) [reply examples/axes_grid1/demo_colorbar_with_axes_divider.py]: Not convinced it's an improvement in legibility...

File: examples/event_handling/zoom_window.py (hunk 1 above)
Comment (anntzer) [reply examples/event_handling/zoom_window.py]: fixed

File: examples/event_handling/lasso_demo.py [hunk 6]
```diff
@@ -10,13 +10,11 @@
...
 usable as is).  There will be some refinement of the API.
 """
-from matplotlib.widgets import Lasso
-from matplotlib.collections import RegularPolyCollection
-from matplotlib import colors as mcolors, path
 
-import matplotlib.pyplot as plt
-from numpy import nonzero
-from numpy.random import rand
+from matplotlib import colors as mcolors, path, pyplot as plt
```
[the remark this answers is not in the document]
Comment (anntzer) [reply examples/event_handling/lasso_demo.py]: disagree, but still split it out

File: examples/units/evans_test.py (hunk 2 above)
Comment (anntzer) [reply examples/units/evans_test.py]: tick_params can set labelrotation, but not horizontalalignment (and I don't think it's worth changing that, IMO setp actually does this job perfectly well...).

File: examples/units/evans_test.py (hunk 3 above)
Comment (anntzer) [reply examples/units/evans_test.py]: ditto

File: examples/user_interfaces/embedding_in_gtk_sgskip.py (hunk 4 above)
Comment (anntzer) [reply examples/user_interfaces/embedding_in_gtk_sgskip.py]: See discussion at https://github.com/matplotlib/matplotlib/pull/7562#issuecomment-264775106 which I now agree with: I still believe spaces are in general good, but in some cases no spaces read better.
See also https://www.python.org/dev/peps/pep-0008/#other-recommendations ("use your own judgment", hehe).

**Pull request merged**